rootfs/templates/wrapper-body.sh

```
@@ -514,36 +572,42 @@
-		CONTAINER_ID=$(docker run --detach --init --name "${DOCKER_NAME}" "${DOCKER_ARGS[@]}" ${GEODESIC_DOCKER_EXTRA_ARGS} ${DOCKER_IMAGE} /usr/local/sbin/shell-monitor)
+		CONTAINER_ID=$(docker run --detach --init --name "${DOCKER_NAME}" "${DOCKER_LAUNCH_ARGS[@]}" "${DOCKER_EXEC_ARGS[@]}" ${GEODESIC_DOCKER_EXTRA_ARGS} ${DOCKER_IMAGE} /usr/local/sbin/shell-monitor)
 		echo "# Started session ${CONTAINER_ID:0:12}. Starting shell via \`docker exec\`..."
-		docker exec -it --detach-keys "ctrl-^,ctrl-[,ctrl-@" --env GEODESIC_HOST_CWD="${GEODESIC_HOST_CWD}" "${DOCKER_NAME}" /bin/bash -l $*
+		if [ $# -eq 0 ]; then
+			set -- "/bin/bash" "-l" "$@"
+		fi
+
+		[ -t 0 ] && DOCKER_EXEC_ARGS+=(-it)
+		# We set unusual detach keys because (a) the default first char is ctrl-p, which is used for command history,
+		# and (b) if you detach from the shell, there is no way to reattach to it, so we want to effectively disable detach.
+		docker exec --env G_HOST_PID=$$ --detach-keys "ctrl-^,ctrl-[,ctrl-@" "${DOCKER_EXEC_ARGS[@]}" "${DOCKER_NAME}" $*
```

Shellcheck (0.10.0)

[warning] Use "$@" (with quotes) to prevent whitespace problems.

(SC2048)